src/lib/components/Segment.svelte

```diff
@@ -105,12 +105,9 @@
     border-radius: var(--border-radius);
     overflow: hidden;
 
-    width: 100%;
-
-    @include media.min-width(medium) {
-      width: var(--segment-min-width, var(--segment-width, fit-content));
-      --segment-min-width: calc(var(--segment-button-width) * var(--segments));
-    }
+    box-sizing: border-box;
+    width: var(--segment-min-width, var(--segment-width, fit-content));
```

Comment:
Why do you need a fallback for `--segment-min-width` if it's defined below?

Reply:
I think the declaration place of `--segment-min-width` is not important if in the same scope. If the `--segment-button-width` is not declared (in nns-dapp for example), the auto-size mechanism is being applied.